src/main/java/formflow/library/data/Submission.java

```diff
@@ -142,21 +157,36 @@ public void mergeFormDataWithSubflowIterationData(String subflowName, Map<String
   }
 
   /**
-   * Removes any data in an interation that has "iterationIsComplete" set to "false" and has a {@code uuid} not equal to the
-   * {@code currentUuid} provided.
+   * Sets the 'iterationIsComplete' status for a given iteration.
    *
-   * @param subflowName subflow that we are checking the iteration data for, not null
-   * @param currentUuid The current uuid being worked on, not null. The data associated with this uuid will not be deleted.
+   * @param subflow       the subflow the iteration is a part of
+   * @param iterationUuid the uuid that identifies the iteration to update
+   * @param newStatus     value of new complete status [true/false]
    */
-  public void removeIncompleteIterations(String subflowName, String currentUuid) {
-    List<Map<String, Object>> toRemove = new ArrayList<>();
-    List<Map<String, Object>> subflow = (List<Map<String, Object>>) inputData.get(subflowName);
-    subflow.forEach(iteration -> {
-      if (iteration.get("iterationIsComplete").equals(false) && !iteration.get("uuid").equals(currentUuid)) {
-        toRemove.add(iteration);
-      }
-    });
-    subflow.removeAll(toRemove);
+  public void setIterationIsCompleteStatus(String subflow, String iterationUuid, boolean newStatus) {
+    Map<String, Object> iterationData = getSubflowEntryByUuid(subflow, iterationUuid);
+    if (iterationData == null) {
+      log.warn("No iteration data found for subflow '{}' with UUID '{}'", subflow, iterationUuid);
+      return;
+    }
+    iterationData.put(ITERATION_IS_COMPLETE_KEY, newStatus);
+  }
+
+  /**
+   * Returns the status of whether the given iteration is complete.
+   *
+   * @param subflow       the subflow the iteration is a part of
+   * @param iterationUuid the iteration to find the completion status of
+   * @return Boolean of true/false, maybe be null if the status is unset
+   */
+  public Boolean getIterationIsCompleteStatus(String subflow, String iterationUuid) {
+    Map<String, Object> iterationData = getSubflowEntryByUuid(subflow, iterationUuid);
+    if (iterationData == null) {
+      log.warn("Iteration completion status request, but iteration with id '{}' not found.", iterationUuid);
+      return null;
+    }
+
+    return (Boolean) iterationData.get(ITERATION_IS_COMPLETE_KEY);
```

Comment:
I added a setter and getter for this flag, so we could easily retrieve it from an iteration.